src/flamenco/runtime/fd_runtime.c

```diff
@@ -1315,6 +1315,135 @@ fd_runtime_block_verify_ticks( fd_blockstore_t * blockstore,
   return FD_BLOCK_OK;
 }
 
+/* MAP_KEY_NULL cannot be all 0s because it is the address of the Solana transfer system program */
+static fd_acct_addr_t fd_acct_addr_null = {.b={0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF}};
+
+#define MAP_NAME              fd_txn_writes
+#define MAP_KEY_T             fd_acct_addr_t
+#define MAP_T                 fd_txn_writes_t
+#define MAP_HASH_T            ulong
+#define MAP_KEY_NULL          (fd_acct_addr_null)
+#define MAP_KEY_EQUAL(k0,k1)  (0==memcmp((k0).b,(k1).b,32))
+#define MAP_KEY_HASH(key)     fd_hash( FD_TXN_CONFLICT_MAP_SEED, key.b, 32 )
+#define MAP_KEY_INVAL(k)      (0==memcmp(&fd_acct_addr_null, (k).b, 32))
+#define MAP_KEY_EQUAL_IS_SLOW 0
+#define MAP_MEMOIZE           0
+
+#include "../../util/tmpl/fd_map_dynamic.c"
+
+ulong
+fd_runtime_microblock_verify_read_write_conflicts( fd_txn_p_t *      txns,
+                                                   ulong             txn_cnt,
+                                                   fd_txn_writes_t * acct_map,
+                                                   fd_funk_t *       funk,
+                                                   fd_funk_txn_t *   funk_txn ) {
+  ulong err=FD_TXN_CONFLICT_OK;
+
+  /* Insert all writable accounts into acct_map and detect write-write conflicts */
+  for( ulong i=0; i<txn_cnt; i++ ) {
+    fd_txn_p_t *                    txn = txns + i;
+    fd_acct_addr_t * account_keys = (fd_acct_addr_t *)fd_type_pun( (void*)fd_txn_get_acct_addrs( TXN(txn), txn->payload ) );
+
+    /* Iterate all writable accounts within the txn binary */
+    for( fd_txn_acct_iter_t j=fd_txn_acct_iter_init( TXN(txn), FD_TXN_ACCT_CAT_WRITABLE );
+         j!=fd_txn_acct_iter_end();
+         j=fd_txn_acct_iter_next( j ) ) {
+      fd_acct_addr_t * writable_acc = &account_keys[ fd_txn_acct_iter_idx( j ) ];
+
+      if( FD_UNLIKELY( fd_txn_writes_query( acct_map, *writable_acc, NULL ) ) ) {
+        err = FD_TXN_CONFLICT_WRITE_WRITE;
+        goto cleanup;
```

Comment:
the control flow is confusing to have both `err` that is eventually returned and `goto` -- can you avoid using `goto` and just return? we generally avoid using `goto` (the exception is a stacked `goto` pattern for initialization failures)

instead you can just clear the map at the start of the function